### PR TITLE
Xdg thumbnailer rebased

### PR DESCRIPTION
**Pull request by azubieta**
rebase of pr #578

### CODE REVIEW

File: include/appimage/libappimage.h [hunk 1]
```diff
@@ -26,6 +26,10 @@ int appimage_register_in_system(const char *path, gboolean verbose);
 /* Unregister an AppImage in the system */
 int appimage_unregister_in_system(const char *path, gboolean verbose);
 
+/* Create AppImage thumbnail according to
+ * https://specifications.freedesktop.org/thumbnail-spec/0.8.0/index.html
+ */
+void create_thumbnail(const gchar * appimage_file_path);
```
Comment (TheAssassin) [inline include/appimage/libappimage.h]: How about `const gchar const*`?

File: src/getsection.c [hunk 2]
```diff
@@ -7,7 +7,7 @@
 #include "light_elf.h"
 
 /* Return the offset, and the length of an ELF section with a given name in a given ELF file */
-int get_elf_section_offset_and_lenghth(char* fname, char* section_name, unsigned long *offset, unsigned long *length)
+int get_elf_section_offset_and_lenghth(const char* fname, char* section_name, unsigned long *offset, unsigned long *length)
```
Comment (TheAssassin) [inline src/getsection.c]: How about `const char const*`?

File: src/shared.c [hunk 3]
```diff
@@ -64,7 +65,7 @@
 
 char *vendorprefix = "appimagekit";
 
-void set_executable(char *path, gboolean verbose)
+void set_executable(const char *path, gboolean verbose)
```
Comment (TheAssassin) [inline src/shared.c]: How about `const char const*`?

File: src/shared.c [hunk 4]
```diff
@@ -98,7 +99,7 @@ gchar* replace_str(const gchar *src, const gchar *find, const gchar *replace){
 /* Return the md5 hash constructed according to
  * https://specifications.freedesktop.org/thumbnail-spec/thumbnail-spec-latest.html#THUMBSAVE
  * This can be used to identify files that are related to a given AppImage at a given location */
-char * get_md5(char *path)
+char * get_md5(const char *path)
```
Comment (TheAssassin) [inline src/shared.c]: How about `const char const*`?

File: src/shared.c [hunk 5]
```diff
@@ -121,7 +122,7 @@ char * get_md5(char *path)
  * Check libgnomeui/gnome-thumbnail.h for actually generating thumbnails in the correct
  * sizes at the correct locations automatically; which would draw in a dependency on gdk-pixbuf.
  */
-char * get_thumbnail_path(char *path, char *thumbnail_size, gboolean verbose)
+char * get_thumbnail_path(const char *path, char *thumbnail_size, gboolean verbose)
```
Comment (TheAssassin) [inline src/shared.c]: How about `const char const*`?

File: src/shared.c [hunk 6]
```diff
@@ -201,7 +202,7 @@ void move_icon_to_destination(gchar *icon_path, gboolean verbose)
 }
 
 /* Check if a file is an AppImage. Returns the image type if it is, or -1 if it isn't */
-int check_appimage_type(char *path, gboolean verbose)
+int check_appimage_type(const char *path, gboolean verbose)
```
Comment (TheAssassin) [inline src/shared.c]: How about `const char const*`?

File: src/shared.c [hunk 7]
```diff
@@ -255,7 +256,27 @@ static gchar *get_file_extension(const gchar *filename)
     return extension;
 }
 
-/* Find files in the squashfs matching to the regex pattern. 
+void squash_extract_inode_to_file(const sqfs *fs, sqfs_inode *inode, const gchar *dest) {// Read the file in chunks
+    off_t bytes_already_read = 0;
+    sqfs_off_t bytes_at_a_time = 64*1024;
```
Comment (TheAssassin) [inline src/shared.c]: This value should be `const`, maybe even `const inline`.

File: src/shared.c [hunk 8]
```diff
@@ -397,7 +401,7 @@ gboolean g_key_file_load_from_squash(sqfs *fs, char *path, GKeyFile *key_file_st
 }
 
 /* Write a modified desktop file to disk that points to the AppImage */
-void write_edited_desktop_file(GKeyFile *key_file_structure, char* appimage_path, gchar* desktop_filename, int appimage_type, char *md5, gboolean verbose){
+void write_edited_desktop_file(GKeyFile *key_file_structure, const char* appimage_path, gchar* desktop_filename, int appimage_type, char *md5, gboolean verbose){
```
Comment (TheAssassin) [inline src/shared.c]: How about `const char const*`?

File: src/shared.c [hunk 9]
```diff
@@ -585,7 +589,7 @@ void write_edited_desktop_file(GKeyFile *key_file_structure, char* appimage_path
 }
 
 /* Register a type 1 AppImage in the system */
-bool appimage_type1_register_in_system(char *path, gboolean verbose)
+bool appimage_type1_register_in_system(const char *path, gboolean verbose)
```
Comment (TheAssassin) [inline src/shared.c]: How about `const char const*`?

File: include/appimage/libappimage.h [hunk 10]
```diff
@@ -26,6 +26,9 @@ int appimage_register_in_system(const char *path, gboolean verbose);
 /* Unregister an AppImage in the system */
 int appimage_unregister_in_system(const char *path, gboolean verbose);
 
+/* Extract a given file from the appimage following the symlinks until a concrete file is found */
+void extract_file_following_symlinks(const char *appimage_file_path, const char *file_path, const char *target_dir);
```
Comment (azubieta) [inline include/appimage/libappimage.h]: Required by the KDE thumbnailer useful to others who aim to extract a single file from an Appimage.

**Comment by TheAssassin** [comment]
Just for completion, this does not yet involve the KDE thumbnailer, as far as I can see. I'll update the branch and will try to merge it.

**Comment by azubieta** [comment]
In fact i don't think that the appimage kit should include a KDE thumbnailer as it is highly dependent and specific for kde we could keep it in a separated project.

**Comment by TheAssassin** [comment]
Thanks a lot for your efforts. We'll discuss the KDE aspects in a separate thread, I'd say.

**Pull request merged**